lib/launchers/amz.rb

```diff
@@ -471,6 +480,63 @@ def get_regions
       client_ec2.describe_regions.to_a[0][0]
     end
 
+    def default_zone
+      config[:install_base_domain].sub(/\.$/,"") + "."
+    end
+
+    def default_zone_id
+      unless @default_zone_id
+        zone ||= client_r53.list_hosted_zones.hosted_zones.find { |z|
+          z.name == default_zone
+        }
+        unless zone
+          raise "can't find the configured default zone in AWS"
+        end
+        @default_zone_id = zone.id.sub(%r{.*/(.*)$}, "\\1")
+      end
+      return @default_zone_id
+    end
+
+    # @param [Hash] change might be something like
+    #   [{
+    #     :action=>"UPSERT",
+    #     :resource_record_set=> {
+    #       :name=>"myhost.qe.devcluster.openshift.com",
+    #       :resource_records=> [{:value=>"192.0.2.44"}],
+    #       :ttl=>60,
+    #       :type=>"A"
+    #     }
+    #   }]
+    # @see https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/Route53/Client.html#change_resource_record_sets-instance_method
+    def change_resource_record_sets(zone_id: nil, changes:)
```

Comment:
Is this method waiting for DNS propagation before returns? I'm asking because normal calling of change-resource-record-set is ASYNC - you get an ID of change and then you have to call 'aws wait resource-record-sets-changed --id ...' to hold execution until the change is propagated (usually it takes couple of seconds (10-30) in common situation).

Probably we should extend this impl. with the attribute whether we want to wait until the propagation finishes or not.

Reply:
It does not presently do so. It can be extended. Is it needed right now for installation?

If it works now, I'd suggest to merge as is and implement some param like `wait: nil`. When `nil` or negative number, then we do not wait and return ID of request. If it is a positive integerm then we wait for up to that number of seconds.

Please tell me.

Reply:
Just only one concern with a negative number - I was thinking of "-1" as a value indicating - I'm willing to wait until it is propagated whatever time it needs (including indefinitely). In your case this needs to be handled on user side however I prefer to leave it on API side...

So what about:
* nil - don't like to wait, provide me an ID aka ASYNC mode - like now (in that case we need another `amz` helper - wait-for ID finished)
* positive number - wait at most specified amount of time (how to indicate we finished/don't finish in time limit? - exception OutOfTime for ex.)
* "-1" - wait until propagated (including counting with unlimited amount of time)

Reply:
agreed